landlab/components/uniform_precip/generate_uniform_precip.py

```diff
@@ -26,13 +27,17 @@ class PrecipitationDistribution(Component):
 
     Construction::
 
-        PrecipitationDistribution(mean_storm_duration=0.0,
+        PrecipitationDistribution(grid=None,
+                                  mean_storm_duration=0.0,
                                   mean_interstorm_duration=0.0,
                                   mean_storm_depth=0.0, total_t=0.0,
                                   delta_t=0.0, random_seed=0)
 
     Parameters
     ----------
+    grid : ModelGrid or None
+        A Landlab grid (optional). If provided, storm intensities will be
+        stored as a grid scalar field as the component simulates storms.
     mean_storm_duration : float
```

Comment:
An anticipated forward compatibility question.

If in the future we make one of a few variants of spatially variable precipitation generator (e.g. the one @SiccarPoint discussed on the LL call, someone else talked about an orographic precipitation component..., I was just talking to someone at NCED about a very simple random number generator based one that I can't remember if someone else is thinking about making) then should we make sure that the name for this variable is appropriate for both OR give this variable a name that is unique to spatially uniform precipitation. 

That is, in the future, would a component that used rainfall flux look for `rainfall__flux` at grid and at node, and raise an error if both existed, and know that if it was provided at grid, that meant it was uniform, and if at node, it is spatially variable.